### PR TITLE
Remove doctree::Import

### PR DESCRIPTION
**Pull request by CraftSpider**
Per the title. Part of cleaning up doctree

### CODE REVIEW

File: src/librustdoc/clean/mod.rs [hunk 1]
```diff
@@ -2172,105 +2174,101 @@ fn clean_extern_crate(
...
+            if let Some(mut items) = inline::try_inline(
+                cx,
+                cx.tcx.parent_module(import.hir_id).to_def_id(),
+                path.res,
+                name,
+                Some(import.attrs),
+                &mut visited,
+            ) {
+                items.push(Item::from_def_id_and_parts(
+                    cx.tcx.hir().local_def_id(import.hir_id).to_def_id(),
```
Comment (jyn514) [inline src/librustdoc/clean/mod.rs:2255]: nit: you use this twice, I would make it a local variable

**Review by jyn514 — APPROVED** [review]
This is awesome, thank you :heart:

**Comment by jyn514** [comment]
```
---- [rustdoc-json] rustdoc-json/nested.rs stdout ----

error: compare failed!
status: exit code: 1
command: "/usr/bin/python2.7" "/checkout/src/test/rustdoc-json/compare.py" "/checkout/src/test/rustdoc-json/nested.expected" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-json/nested/nested.json" "/checkout/src/test/rustdoc-json"
stdout:
------------------------------------------
checking that /checkout/src/test/rustdoc-json/nested.expected is a logical subset of /checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-json/nested/nested.json

------------------------------------------
stderr:
------------------------------------------
Traceback (most recent call last):
  File "/checkout/src/test/rustdoc-json/compare.py", line 129, in <module>
    main(sys.argv[1], sys.argv[2], normalize(sys.argv[3]))
  File "/checkout/src/test/rustdoc-json/compare.py", line 119, in main
    check_subset(expected_main, actual_main, base_dir)
  File "/checkout/src/test/rustdoc-json/compare.py", line 82, in check_subset
    _check_subset(expected_main["root"], actual_main["root"], [])
  File "/checkout/src/test/rustdoc-json/compare.py", line 76, in _check_subset
    expected_index.get(expected, {}), actual_index.get(actual, {}), trace
  File "/checkout/src/test/rustdoc-json/compare.py", line 58, in _check_subset
    _check_subset(expected[key], actual[key], new_trace)
  File "/checkout/src/test/rustdoc-json/compare.py", line 58, in _check_subset
    _check_subset(expected[key], actual[key], new_trace)
  File "/checkout/src/test/rustdoc-json/compare.py", line 72, in _check_subset
    _check_subset(expected, actual, new_trace)
  File "/checkout/src/test/rustdoc-json/compare.py", line 76, in _check_subset
    expected_index.get(expected, {}), actual_index.get(actual, {}), trace
  File "/checkout/src/test/rustdoc-json/compare.py", line 58, in _check_subset
    _check_subset(expected[key], actual[key], new_trace)
  File "/checkout/src/test/rustdoc-json/compare.py", line 58, in _check_subset
    _check_subset(expected[key], actual[key], new_trace)
  File "/checkout/src/test/rustdoc-json/compare.py", line 72, in _check_subset
    _check_subset(expected, actual, new_trace)
  File "/checkout/src/test/rustdoc-json/compare.py", line 76, in _check_subset
    expected_index.get(expected, {}), actual_index.get(actual, {}), trace
  File "/checkout/src/test/rustdoc-json/compare.py", line 58, in _check_subset
    _check_subset(expected[key], actual[key], new_trace)
  File "/checkout/src/test/rustdoc-json/compare.py", line 44, in _check_subset
    "expected type `{}`, got `{}`".format(expected_type, actual_type), trace
  File "/checkout/src/test/rustdoc-json/compare.py", line 25, in __init__
    super().__init__("{}: {}".format(trace, msg))
TypeError: super() takes at least 1 argument (0 given)
```

@aDotInTheVoid do you recognize what's going on here?

File: src/test/rustdoc-json/nested.expected [hunk 2]
```diff
@@ -41,8 +41,8 @@
       "inner": {
         "is_crate": false,
         "items": [
-          "0:7",
-          "0:4"
+          "0:4",
+          "0:7"
```
Comment (jyn514) [inline src/test/rustdoc-json/nested.expected:45]: Can you add a comment to `compare.py` saying that the error message is misleading and might be related to ordering?

Comment (CraftSpider) [reply src/test/rustdoc-json/nested.expected:45]: Yeah, will do

**Comment by jyn514** [comment]
@bors r+

Thanks!

**Comment by CraftSpider** [comment]
Rebased to fix conflicts

**Comment by CraftSpider** [comment]
@jyn514 Could you re-approve?

**Comment by jyn514** [comment]
Oops, thanks for the ping.

@bors r+

**Comment by aDotInTheVoid** [comment]
> @aDotInTheVoid do you recognize what's going on here?

For posterity its a python2/3 [thing](https://rust-lang.zulipchat.com/#narrow/stream/266220-rustdoc/topic/Removing.20doctree.20stuff/near/222781245)

**Pull request merged**